### PR TITLE
Account class to fetch account information

### PR DESCRIPTION
**Pull request by coco-sha**

### CODE REVIEW

File: packages/polar/sample-project/polar.config.js [hunk 1]
```diff
@@ -1,3 +1,11 @@
+const accounts = [
```
Comment (udit-gulati) [inline packages/polar/sample-project/polar.config.js:1]: Add one more account here please.

File: packages/polar/src/types.ts [hunk 2]
```diff
@@ -23,6 +23,11 @@ export interface Account {
   mnemonic: string
 }
 
+export interface UserAccount {
+  account: Account
+  getBalance: () => Promise<any>
```
Comment (udit-gulati) [inline packages/polar/src/types.ts:28]: use balance type here

File: packages/polar/sample-project/scripts/sample-script.js [hunk 3]
```diff
@@ -1,18 +1,17 @@
...
   const contract = new Contract('sample-project', runtimeEnv);
   await contract.parseSchema();
 
-  const account = runtimeEnv.config.networks.testnet.accounts[0];
-  const deploy_response = await contract.deploy(account);
+  const deploy_response = await contract.deploy(contract_owner.account);
   console.log(deploy_response);
 
-  const contract_info = await contract.instantiate({"count": 102}, "deploy test", account);
+  const contract_info = await contract.instantiate({"count": 102}, "deploy test", contract_owner.account);
```
Comment (udit-gulati) [inline packages/polar/sample-project/scripts/sample-script.js:11]: We can use type `(Account | UserAccount)` here and then only pass `contract_owner`.

File: packages/polar/src/lib/account.ts [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+  constructor (account: Account, env: PolarRuntimeEnvironment) {
+    this.account = account;
+    this.client = getClient(env.network);
+  }
+
+  async getAccountInfo (): Promise<any> {
+    return await this.client.getAccount(this.account.address);
+  }
+
+  async getBalance (): Promise<any> {
```
Comment (udit-gulati) [inline packages/polar/src/lib/account.ts:21]: use a type here

File: packages/polar/src/lib/account.ts [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+  }
+
+  async getSequence (): Promise<any> {
+    const info = await this.client.getAccount(this.account.address);
+    return info?.sequence;
+  }
+}
+
+export function getAccountByName (name: string, env: PolarRuntimeEnvironment): UserAccount {
+  console.log(env.network);
```
Comment (udit-gulati) [inline packages/polar/src/lib/account.ts:43]: Fix this log

**Review by udit-gulati — CHANGES_REQUESTED** [review]

**Review by hunny98 — APPROVED** [review]

File: packages/polar/src/lib/account.ts (hunk 4 above)
Comment (hunny98) [reply packages/polar/src/lib/account.ts:21]: Type 'readonly Coin[] | undefined' is not assignable to type 'string'.
Type 'undefined' is not assignable to type 'string'.ts(2322)
@udit-gulati ?

File: packages/polar/src/lib/account.ts (hunk 5 above)
Comment (hunny98) [reply packages/polar/src/lib/account.ts:43]: could you please elaborate? @udit-gulati

**Review by udit-gulati — APPROVED** [review]

**Pull request merged**